upf/upf_pfcp_api.c

```diff
@@ -736,14 +739,38 @@ handle_f_teid (upf_session_t * sx, upf_main_t * gtm, pfcp_pdi_t * pdi,
     }
   else
     {
-      // CH == 0
+      /* CH == 0, check for conflicts with other sessions */
+
+      if (pdi->f_teid.flags & F_TEID_V4)
+	sidx =
+	  teid_v4_lookup_session_index (gtm, pdi->f_teid.teid,
+					&pdi->f_teid.ip4);
+      else if (pdi->f_teid.flags & F_TEID_V6)
+	sidx =
+	  teid_v6_lookup_session_index (gtm, pdi->f_teid.teid,
+					&pdi->f_teid.ip6);
+      if (sidx != ~0 && sidx != sx - gtm->sessions)
+	return -1;
+
       process_pdr->pdi.teid = pdi->f_teid;
     }
 
   return 0;
 
 }
 
+static int
+validate_ue_ip (upf_session_t * sx, upf_nwi_t * nwi,
+		pfcp_ue_ip_address_t * ue_addr)
+{
+  upf_main_t *gtm = &upf_main;
+  const dpo_id_t *dpo = ue_addr->flags & IE_UE_IP_ADDRESS_V4 ?
```

Comment:
Sadly, UE IP can contain both IPv4 *and* IPv6. There is no restriction in TS 29.244 on the combination of the flags.

The only thing that could interpreted as a restriction is this:

>  It contains a source or destination IP address

Its in singular and that would im indicate the two addresses are not expected.

A quick search shows that the other places that deal with UE IP Address IEs except instances with both IP versions set.

Reply:
Fixed, thanks